aiohttp/client.py

```diff
@@ -792,10 +791,14 @@ async def _ws_connect(
             extstr = ws_ext_gen(compress=compress)
             real_headers[hdrs.SEC_WEBSOCKET_EXTENSIONS] = extstr
 
+        # For the sake of backward compatibility, if user passes in None, convert it to True
```

Comment:
Are we doing the backwards compatible approach here, and then removing it with a second PR, or just removing it in this PR? Currently, you're doing both, so it's _not_ backwards compatible and can't be backported, and you have redundant code like this so it doesn't make sense to merge on its own.

Reply:
I am leaning towards doing the backwards compatible approach and keeping it. :)

Reply:
TBH, as long as we have a deterministic behavior for ssl=None, I am not a big fan of removing it from allowed type.

Reply:
It will provide better consistency and v4 will likely still be some time away, so there's plenty of time to migrate.
I think we just need the DeprecationWarning in this PR then, and it should be good to go.